cmd/kube-apiserver/app/options/options.go

```diff
@@ -109,6 +110,65 @@ func NewServerRunOptions() *ServerRunOptions {
 	return &s
 }
 
+// ComponentConfigz produces a componentconfig object from server options.
+func (s *ServerRunOptions) ComponentConfigz() componentconfig.APIServerConfiguration {
+	serviceClusterIPRange := s.ServiceClusterIPRange.String()
+	serviceNodePortRange := s.ServiceNodePortRange.String()
+
+	componentconfig := componentconfig.APIServerConfiguration{
+		AdmissionControl:            s.GenericServerRunOptions.AdmissionControl,
+		AdmissionControlConfigFile:  s.GenericServerRunOptions.AdmissionControlConfigFile,
+		AdvertiseAddress:            s.GenericServerRunOptions.AdvertiseAddress.String(),
+		CorsAllowedOriginList:       s.GenericServerRunOptions.CorsAllowedOriginList,
+		ExternalHost:                s.GenericServerRunOptions.ExternalHost,
+		MaxRequestsInFlight:         s.GenericServerRunOptions.MaxRequestsInFlight,
+		MaxMutatingRequestsInFlight: s.GenericServerRunOptions.MaxMutatingRequestsInFlight,
+		MinRequestTimeout:           s.GenericServerRunOptions.MinRequestTimeout,
+		TargetRAMMB:                 s.GenericServerRunOptions.TargetRAMMB,
+		WatchCacheSizes:             s.GenericServerRunOptions.WatchCacheSizes,
+		AuditLogOptions: componentconfig.APIServerAuditLogOptions{
+			Path:       s.Audit.Path,
+			MaxAge:     s.Audit.MaxAge,
+			MaxBackups: s.Audit.MaxBackups,
+			MaxSize:    s.Audit.MaxSize,
+		},
+		EnableProfiling:           s.Features.EnableProfiling,
+		EnableContentionProfiling: s.Features.EnableContentionProfiling,
+		EnableSwaggerUI:           s.Features.EnableSwaggerUI,
+		StorageConfig: componentconfig.APIServerEtcdConfiguration{
+			Type:       s.Etcd.StorageConfig.Type,
+			Prefix:     s.Etcd.StorageConfig.Prefix,
+			ServerList: s.Etcd.StorageConfig.ServerList,
+			KeyFile:    s.Etcd.StorageConfig.KeyFile,
+			CertFile:   s.Etcd.StorageConfig.CertFile,
+			CAFile:     s.Etcd.StorageConfig.CAFile,
+			Quorum:     s.Etcd.StorageConfig.Quorum,
+			DeserializationCacheSize: s.Etcd.StorageConfig.DeserializationCacheSize,
+		},
+		EtcdServersOverrides:    s.Etcd.EtcdServersOverrides,
```

Comment:
rather than second guess the confidentiality of every config field, I'd rather start from the position that apiservers don't publish their active config via an API